packages/jet-brains-integration/src/web-types-generator.ts

```diff
@@ -71,33 +71,46 @@ function getJsProperties(
   };
 }
 
+/**
+ * @param field The Custom Elements Manifest class field to evaluate
+ * @return Whether the Custom Elements Manifest class field is a public property
+ */
+function isPublicProperty(field: schema.ClassField): boolean {
+  // It appears that JetBrains Web Types assumes that all properties are public (TS only) and not
+  // static.
+  return field.static !== true && (field.privacy === 'public' || field.privacy === undefined)
+}
+
 function getWebTypeProperties(
   component: Component,
-  typesSrc = "types"
-): WebTypeAttribute[] {
+  typesSrc = 'type'
+): WebTypeJsProperty[] {
   return (
-    ((component.attributes || component.members) as schema.Attribute[])?.map(
-      (attr) => {
+    (component.members?.filter((member) => member.kind === 'field') as schema.ClassField[])
+      .filter(isPublicProperty)
+      .map((field) => {
```

Comment:
It looks like the `.filter` and `.map` need to be prefixed with a `?`.

Reply:
If `component.members` is null or undefined, the entire chain of function calls on the array will be ignored and `undefined` will be returned. The `?.` operator won't do anything after the first `filter()` because that function never returns null or undefined, but it is possible that it could return an empty array. If you are concerned about that, I can update this and the other instances to return `undefined` if the final array is empty.

Reply:
No, you're right this one should be fine.

Reply:
The analyzer is throwing an error in this method when calling `filter`. You should be able to see the error when you run `turbo analyze`.

Reply:
Oh I see now, it's the parentheses that are needed for the cast. If `component.members` is `undefined`, it is returned out of the parentheses used for the cast, so the second `filter()` could be called on `undefined`. Pushing a fix shortly.

Reply:
Just pushed a fix, `turbo analyze` passes locally for me now!